server/services/reportService.ts

```diff
@@ -0,0 +1,188 @@
+import moment from 'moment'
+
+import type IncidentClient from '../data/incidentClient'
+import type { ReportSummary, IncompleteReportSummary } from '../data/incidentClientTypes'
+import type { SystemToken, OffenderService } from '../types/uof'
+
+import logger from '../../log'
+import { isNilOrEmpty } from '../utils/utils'
+import { check as getReportStatus } from './reportStatusChecker'
+import { InTransaction } from '../data/dataAccess/db'
+
+interface NamesByOffenderNumber {
+  [offenderNo: string]: string
+}
+
+export interface IncidentSummary {
+  id: number
+  bookingId: number
+  incidentdate: Date
+  staffMemberName: string
+  offenderName: string
+  offenderNo: string
+  status: string
+}
+
+const toReport = (namesByOffenderNumber: NamesByOffenderNumber) => (
+  reportSummary: ReportSummary | IncompleteReportSummary
+): IncidentSummary => ({
+  id: reportSummary.id,
+  bookingId: reportSummary.bookingId,
+  incidentdate: reportSummary.incidentDate,
+  staffMemberName: reportSummary.reporterName,
+  offenderName: namesByOffenderNumber[reportSummary.offenderNo],
+  offenderNo: reportSummary.offenderNo,
+  status: reportSummary.status,
+})
+
+export default class ReportService {
+  constructor(
+    private readonly incidentClient: IncidentClient,
+    private readonly elite2ClientBuilder,
+    private readonly involvedStaffService,
+    private readonly notificationService,
+    private readonly offenderService: OffenderService,
+    private readonly inTransaction: InTransaction,
+    private readonly systemToken: SystemToken
+  ) {}
+
+  private async getOffenderNames(username, incidents): Promise<NamesByOffenderNumber> {
+    const token = await this.systemToken(username)
+    const offenderNos = incidents.map(incident => incident.offenderNo)
+    return this.offenderService.getOffenderNames(token, offenderNos)
+  }
+
+  getCurrentDraft(userId, bookingId) {
+    return this.incidentClient.getCurrentDraftReport(userId, bookingId)
+  }
+
+  getReportStatus(report) {
```

Comment:
Why does this need wrapping like this?

Reply:
the function is imported from another module. mixing in this functionality as it feels like it belongs here and its convenient to mock out the entire service when included as a dependency rather than 2 separate parts.